tomosegmemtv/protocols/protocol_annotate_membranes.py

```diff
@@ -94,17 +151,31 @@ def _summary(self):
                 summary.append('All segmentations have been already annotated.')
         return summary
 
+    def _validate(self):
+        error = []
+        # This is a tolerance in the sampling rate to ensure that tomoMask and tomograms have similar pixel size
+        tolerance = 0.001
+        if self.inputTomos.get():
+            if abs(self.inputTomos.get().getSamplingRate() - self.inputTomoMasks.get().getSamplingRate()) > tolerance:
+                error.append('The sampling rate of the tomograms does not match the sampling rate of the input masks')
+
+        return error
+
     # --------------------------- UTIL functions -----------------------------------
 
     def _initialize(self):
-        self._tomoList = [tomo.clone() for tomo in self.inputTomoMasks.get().iterItems()]
-        self._provider = MembAnnotatorProvider(self._tomoList, self._getExtraPath(), 'membAnnotator')
+        import time
+        time.sleep(12)
+        self._tomoMaskDict = {tomoMask.getTsId(): tomoMask.clone() for tomoMask in self.inputTomoMasks.get().iterItems()}
+        if self.inputTomos.get():
+            self._tomoDict = {tomo.getTsId(): tomo.clone() for tomo in self.inputTomos.get().iterItems()}
```

Comment:
Where is _tomoDict used?

Reply:
It is used in the input, to represent the tomograms by the annotator. It is used for visualization purpose